src/aggregators/direct.jl

```diff
@@ -55,14 +57,54 @@ end
   @inbounds cur_rates[idx] = rate(u,p,t)
   nothing
 end
+##################################################################
 
-@inline function aggregate(aggregator::Direct,u,p,t,end_time,constant_jumps,save_positions)
-  rates = ((c.rate for c in constant_jumps)...)
+############ FunctionWrapper based time to next jump #############
+function time_to_next_jump(u, p, t, rates::Vector{T}, cur_rates) where T
+  prev_rate = zero(t)
+  new_rate  = zero(t)
+  @inbounds for i in 1:length(rates)
+      new_rate     = rates[i](u, p, t)
+      cur_rates[i] = new_rate + prev_rate
+      prev_rate    = cur_rates[i]
+  end
+  @inbounds sum_rate = cur_rates[end]
+  sum_rate,randexp()/sum_rate
+end
+##################################################################
+
+function getRatesAffectsAsTuples(constant_jumps)
+  rates    = ((c.rate for c in constant_jumps)...)
   affects! = ((c.affect! for c in constant_jumps)...)
-  cur_rates = Vector{Float64}(length(rates))
+
+  return rates, affects!
+end
+
+function getRatesAffectsAsFWrappers(u, p, t, constant_jumps)
+  RateWrapper   = FunctionWrappers.FunctionWrapper{typeof(t),Tuple{typeof(u), typeof(p), typeof(t)}}
+  rates         = [RateWrapper(c.rate) for c in constant_jumps]
+  AffectWrapper = FunctionWrappers.FunctionWrapper{Void,Tuple{Any}}
+  affects!      = [AffectWrapper(x->(c.affect!(x);nothing)) for c in constant_jumps]
+
+  return rates, affects!
+end
+
+
+# when to switch from tuples for rates/affects to FunctionWrappers
+const TUPLE_TO_FWRAPPER_CUTOFF = 10
```

Comment:
It feels weird to try and just make this built in. I think that instead it can just be a new aggregator, but with a lot of common code. Which one to use is problem dependent. @alanderos91 seems to have ideas for generalizing the code re-use.

Reply:
Ok, I'm to remove this PR then. Once @alanderos91 finishes revising the interface I can look into sending in updates to enable larger SSAs.